alephium/devnet/deploy.ts

```diff
@@ -0,0 +1,108 @@
+import { mine } from './mine'
+import * as consts from './consts'
+import * as dotenv from "dotenv"
+
+dotenv.config({ path: __dirname+'/../../.env' })
+
+const provider = new NodeProvider("http://localhost:22973")
+
+async function createWallet() {
+    const wallets = await provider.wallets.getWallets()
+    const exists = wallets.some(status => status.walletName == consts.testWalletName)
+    if (exists) {
+        console.log('test wallet already exists')
+        await provider.wallets.postWalletsWalletNameUnlock(consts.testWalletName, { password: consts.testWalletPassword })
+        return
+    }
+
+    await provider.wallets.putWallets({
+        walletName: consts.testWalletName,
+        mnemonic: consts.testWalletMnemonic,
+        password: consts.testWalletPassword
+    })
+    console.log('create test wallet succeed')
+}
+
+async function createLocalTokenPool(
+    wormhole: Wormhole,
+    localTokenId: string,
+    tokenBridgeId: string
+) {
+    let txId = await wormhole.createLocalTokenPool(tokenBridgeId, localTokenId, consts.payer, consts.minimalAlphInContract)
+    let tokenPoolAddress = await getCreatedContractAddress(provider, txId, 0)
+    const tokenPoolId = binToHex(contractIdFromAddress(tokenPoolAddress))
+    console.log('local token id: ' + localTokenId + ', token pool id: ' + tokenPoolId)
+}
+
+async function createWrappedAlphPool(
+    wormhole: Wormhole,
+    tokenBridgeId: string
+) {
+    let txId = await wormhole.createWrappedAlphPool(tokenBridgeId, consts.payer, consts.minimalAlphInContract)
+    let alphPoolAddress = await getCreatedContractAddress(provider, txId, 0)
+    const alphPoolId = binToHex(contractIdFromAddress(alphPoolAddress))
+    console.log('alph pool id: ' + alphPoolId)
+}
+
+async function getToken(
+    signer: NodeWallet,
+    tokenId: string,
+    from: string,
+    amount: bigint
+): Promise<string> {
+    const script = Project.script('tests/get_token.ral')
+    const scriptTx = await script.transactionForDeployment(signer, {
+        initialFields: {
+            sender: from,
+            amount: amount,
+            tokenId: tokenId
+        }
+    })
+    const result = await signer.submitTransaction(scriptTx.unsignedTx, scriptTx.txId)
+    return result.txId
+}
+
+async function deploy() {
+    await createWallet()
+    await Project.build(provider)
+
+    const signer = await testWallet(provider)
+    // deploy the test token first to make sure the contract id is deterministic
+    const testTokenId = await deployTestToken(signer)
+
+    const initGuardianSet = JSON.parse(process.env.INIT_SIGNERS!) as string[]
+    const wormhole = new Wormhole(
+        provider,
+        signer,
+        consts.payer,
+        parseInt(process.env.INIT_CHAIN_ID!),
+        parseInt(process.env.INIT_GOV_CHAIN_ID!),
+        process.env.INIT_GOV_CONTRACT!,
+        initGuardianSet,
+        0,
+        consts.messageFee
+    )
+
+    const contracts = await wormhole.deployContracts("devnet")
```

Comment:
I've also thought about this before, it's technically possible to deploy to multiple groups, but if we deploy to multiple groups, we also need to:

1. Deploy wrapped ALPH on multiple groups
2. Deploy the token wrapper on multiple groups
3. Listen for token bridge messages from multiple groups

And I'm not sure if it will affect the user experience, but maybe we can give it a try.

Reply:
Thanks. If we do not deploy them to multiple groups, what does that mean to UX?

Reply:
Sorry, I thought about it again, and feels like we can completely hide these details, which should have no impact on the user experience.